### PR TITLE
[chaturbate] Fix Chaturbate extractor

### PR DESCRIPTION
**Pull request by einstein95**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixes the Chaturbate extractor reporting "room offline" for rooms which are online, due to changes in the m3u8 location on the page.

### CODE REVIEW

File: youtube_dl/extractor/chaturbate.py [hunk 1]
```diff
@@ -32,7 +44,7 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         m3u8_url = self._search_regex(
-            r'src=(["\'])(?P<url>http.+?\.m3u8.*?)\1', webpage,
+            r'var hlsSource.+? = (["\'])(?P<url>http.+?\.m3u8.*?)\1', webpage,
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: All formats should be extracted.

File: youtube_dl/extractor/chaturbate.py [hunk 2]
```diff
@@ -7,6 +7,18 @@
...
+            'id': 'joshmin3207',
+            'ext': 'mp4',
+            'title': 're:^joshmin3207 [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
+            'age_limit': 18,
+            'is_live': True,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: No need in another test.

File: youtube_dl/extractor/chaturbate.py [hunk 3]
```diff
@@ -31,30 +40,34 @@ def _real_extract(self, url):
...
         webpage = self._download_webpage(url, video_id)
 
-        m3u8_url = self._search_regex(
-            r'src=(["\'])(?P<url>http.+?\.m3u8.*?)\1', webpage,
+        m3u8_url_fast = self._search_regex(
+            r'var hlsSourceFast = (["\'])(?P<url>http.+?\.m3u8)', webpage,
+            'playlist', default=None, group='url')
+        m3u8_url_slow = self._search_regex(
+            r'var hlsSourceSlow = (["\'])(?P<url>http.+?\.m3u8)', webpage,
             'playlist', default=None, group='url')
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: Code duplication.

File: youtube_dl/extractor/chaturbate.py [hunk 4]
```diff
@@ -31,30 +40,34 @@ def _real_extract(self, url):
...
+                if any(p in webpage for p in (
                         self._ROOM_OFFLINE, 'offline_tipping', 'tip_offline')):
                     error = self._ROOM_OFFLINE
             if error:
                 raise ExtractorError(error, expected=True)
             raise ExtractorError('Unable to find stream URL')
 
-        formats = self._extract_m3u8_formats(m3u8_url, video_id, ext='mp4')
+        formats = self._extract_m3u8_formats(m3u8_url_fast, video_id, ext='mp4')
+        formats.append(self._extract_m3u8_formats(m3u8_url_slow, video_id, ext='mp4')[0])
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: Code duplication.

File: youtube_dl/extractor/chaturbate.py [hunk 5]
```diff
@@ -7,6 +7,15 @@
...
     _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.com/(?P<id>[^/?#]+)'
     _TESTS = [{
+        'url': 'https://www.chaturbate.com/surfergirl121/',
+        'info_dict': {
+            'id': 'surfergirl121',
+            'ext': 'mp4',
+            'title': 're:^surfergirl121 [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
+            'age_limit': 18,
+            'is_live': True,
+        },
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: I've already pointed out there is no need in another test since they are broken once room is offline.

**Comment by einstein95** [comment]
Removed the code duplication from adding the hlsSourceSlow URL

File: youtube_dl/extractor/chaturbate.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         m3u8_url = self._search_regex(
-            r'src=(["\'])(?P<url>http.+?\.m3u8.*?)\1', webpage,
+            r'var hlsSourceFast = (["\'])(?P<url>http.+?\.m3u8)', webpage,
```
Comment (dstftw) [inline youtube_dl/extractor/chaturbate.py]: I did not ask to remove **all formats extraction** but to remove **code duplication**.

**Pull request closed without merging**